src/mlir/ast-utils.cc

```diff
@@ -323,4 +336,25 @@ namespace mlir::verona::ASTInterface
     return node.lock()->nodes[0];
   }
 
+  // ================================================= Condition Helpers
+  bool isWhile(::ast::WeakAst ast)
+  {
+    return isA(ast, NodeKind::While);
+  }
+
+  bool isContinue(::ast::WeakAst ast)
+  {
+    return isValue(ast) && isA(ast, NodeKind::Continue);
+  }
+
+  bool isBreak(::ast::WeakAst ast)
+  {
+    return isValue(ast) && isA(ast, NodeKind::Break);
+  }
+
+  ::ast::WeakAst getLoopBlock(::ast::WeakAst ast)
+  {
+    assert(isWhile(ast) && "Bad node");
```

Comment:
Similarly here, it would be good to factor the check out into an isLoop check (so isConditional can be isLoop or isIf), the. This will work automatically when we add other loop kinds.